core/src/main/java/org/opensearch/sql/data/model/ExprDateValue.java

```diff
@@ -67,7 +66,12 @@ public LocalDateTime datetimeValue() {
 
   @Override
   public Instant timestampValue() {
-    return ZonedDateTime.of(date, timeValue(), ZoneId.systemDefault()).toInstant();
+    return ZonedDateTime.of(date, timeValue(), ExprTimestampValue.ZONE).toInstant();
+  }
+
+  @Override
+  public boolean isDateTime() {
+    return true;
```

Comment:
What does this function do?

Reply:
Should this be `isDate()` instead?

Reply:
It is used to activate `compareTo` interface for different datetime types, see ab60013.